tests/playwright-test/reporter-json.spec.ts

```diff
@@ -327,3 +327,25 @@ test.describe('report location', () => {
     expect(fs.existsSync(testInfo.outputPath('foo', 'bar', 'baz', 'my-report.json'))).toBe(true);
   });
 });
+
+test('should report parallelIndex', async ({ runInlineTest }, testInfo) => {
+  const result = await runInlineTest({
+    'tests/a.spec.js': `
+      import { test, expect } from '@playwright/test';
+      const fs = require('fs');
+      test('test 1 passes!', async ({}) => {});
+      test('test 2 fails!', async ({}) => {
+        expect(1 + 1).toBe(3);
+      });
+      test('test 3 passes!', async ({}) => {});
+    `
+  }, { 'workers': '1', 'reporter': 'json' });
```

Comment:
Let's run with 2 workers to cover the case when `parallelIndex` is not `0`, otherwise looks good.